### PR TITLE
Feature/postmodel code completion

### PR DESCRIPTION
**Pull request by maikelRAOW**
* Adds PHPDOC info regarding a bunch of functions
* Adds the static "query" function on postmodel. This returns a WpQueryBuilderModel which should improve code completion in most editors.

### CODE REVIEW

File: src/Content/Post/PostModel.php [hunk 1]
```diff
@@ -131,46 +153,72 @@ public function getContent()
...
+     */
     public function getPostName()
     {
         return $this->wpPost->post_name;
     }
 
+    /**
+     * @return string
+     */
     public function getSlug()
```
Comment (dseegers) [inline src/Content/Post/PostModel.php:167]: public function getSlug(): string

Comment (ChrisRAoW) [reply src/Content/Post/PostModel.php:167]: @dseegers HE'S ALIVE!!!

Comment (maikelRAOW) [reply src/Content/Post/PostModel.php]: Ik hoop dat WP altijd een string terug geeft en niet null. 🙃

**Comment by dseegers** [comment]
@maikelRAOW  I'm not really into the doc blocks myself, since the function or method itself has to indicate what it does. If not, it should be broken down into small methods.

A doc block doesn't guarantee what the function (or method) should do, and that scares me a little bit.

When you make the methods strict (what goes in and out) the method just won't work if you put wrong values in. Ultimately that is very chill.

**Comment by maikelRAOW** [comment]
> @maikelRAOW I'm not really into the doc blocks myself, since the function or method itself has to indicate what it does. If not, it should be broken down into small methods.
> 
> A doc block doesn't guarantee what the function (or method) should do, and that scares me a little bit.
> 
> When you make the methods strict (what goes in and out) the method just won't work if you put wrong values in. Ultimately that is very chill.

I actually agree, I prefer types over docblocks. This PR was originally written so that it could never break anything.
Also we have to support PHP 7 which means we can't use union types yet without PHPdoc. );

**Comment by ChrisRAoW** [comment]
@maikelRAOW Are those for example ": string" after the method necessary / widely supported?

**Comment by maikelRAOW** [comment]
> @maikelRAOW Are those for example ": string" after the method necessary / widely supported?

https://mlocati.github.io/articles/php-type-hinting.html
Supported since PHP 7.1

@dseegers Did a pretty good explanation on why they're useful. 😃

**Pull request merged**